spanner/value.go

```diff
@@ -1032,6 +1087,56 @@ func decodeValue(v *proto3.Value, t *sppb.Type, ptr interface{}) error {
 			return errUnexpectedNumericStr(x)
 		}
 		*p = *y
+	case *NullJSON:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if code == sppb.TypeCode_ARRAY {
+			x, err := getListValue(v)
```

Comment:
I think we should add a check here that the array element type is `JSON`.

Reply:
Fixed.